src/s1reader/s1_annotation.py

```diff
@@ -0,0 +1,497 @@
+                str_elem = elem.find(name_vector).text
+                list_out[i] = float(str_elem)
+
+        elif str_type == 'vector_int':
+            for i,elem in enumerate(element_to_parse):
+                str_elem = elem.find(name_vector).text
+                list_out[i] = np.array([int(strin) for strin in str_elem.split()])
+
+        elif str_type == 'vector_float':
+            for i,elem in enumerate(element_to_parse):
+                str_elem = elem.find(name_vector).text
+                list_out[i] = np.array([float(strin) for strin in str_elem.split()])
+
+        elif str_type == 'str':
+            list_out = element_to_parse[0].find(name_vector).text
+
+        else:
+            raise ValueError(f'Cannot recognize the type of the element: {str_type}')
+
+        return list_out
+
+
+@dataclass
+class CalibrationAnnotation(AnnotationBase):
+    '''Reader for Calibration Annotation Data Set (CADS)'''
+    list_azimuth_time: np.ndarray
+    list_line: list
+    list_pixel: None
+    list_sigma_nought: list
+    list_beta_nought : list
+    list_gamma: list
+    list_dn: list
+
+    @classmethod
+    def from_et(cls, et_in=None):
+        '''Extracts the list of calibration informaton from etree from CADS'''
+        cls.xml_et = et_in
+        cls.list_azimuth_time = cls._parse_vectorlist('calibrationVectorList', 'azimuthTime', 'datetime')
+        cls.list_line = cls._parse_vectorlist('calibrationVectorList', 'line', 'scalar_int')
+        cls.list_pixel = cls._parse_vectorlist('calibrationVectorList', 'pixel', 'vector_int')
+        cls.list_sigma_nought = cls._parse_vectorlist('calibrationVectorList', 'sigmaNought', 'vector_float')
+        cls.list_beta_nought = cls._parse_vectorlist('calibrationVectorList', 'betaNought', 'vector_float')
+        cls.list_gamma = cls._parse_vectorlist('calibrationVectorList', 'gamma', 'vector_float')
+        cls.list_dn = cls._parse_vectorlist('calibrationVectorList', 'dn', 'vector_float')
+
+        return cls
+
+
+
+
+@dataclass
+class NoiseAnnotation(AnnotationBase):
+    '''Reader for Noise Annotation Data Set (NADS) for IW SLC'''
+    # NOTE Schema of the NADS is slightly different before/after IPF version 2.90. Needs to be adaptive in accordance with the version.
+    # The issue above was fixed in further implement of thermal noise correction. A separete PR regarding this will be submitted upon the acceptance of this code.
+    # in ISCE2 code: if float(self.IPFversion) < 2.90:
+    # REF: .../isce2/components/isceobj/Sensor/GRD/Sentinel1.py
+
+    rg_list_azimuth_time: np.ndarray
+    rg_list_line: list
+    rg_list_pixel: list
+    rg_list_noise_range_lut: list
+    az_first_azimuth_line: int
+    az_first_range_sample: int
+    az_last_azimuth_line: int
+    az_last_range_sample: int
+    az_line: np.ndarray
+    az_noise_azimuth_lut: np.ndarray
+
+    @classmethod
+    def from_et(cls,et_in: ET, ipf_version: version.Version = version.parse('3.10')):
```

Comment:
Agreed. In fact, `s1_reader.py` in this PR already assigns the parameter when using it.

s1_reader.py, from line 366
```python

#load the Noise annotation
    noise_annotation_path = annotation_path.replace('annotation/','annotation/calibration/noise-')
    with open_method(noise_annotation_path, 'r') as f_nads:
        tree_nads = ET.parse(f_nads)
        noise_annotation = s1_annotation.NoiseAnnotation.from_et(tree_nads, ipf_version=ipf_version)
```

 I've made the changes accordingly.

Reply:
My apologies. The old snippet was posted above. Please see the updated one. Again, I agree with your suggestion.

Reply:
I see, @seongsujeong . But I think the argument is still optional:

https://github.com/opera-adt/s1-reader/blob/2986152ed793f9fe08c72583f1d2ca5792e3c884/src/s1reader/s1_annotation.py#L178

Or maybe you have a reason to keep this argument optional?

Reply:
@gshiroma I am making changes locally, and have not yet pushed it to my fork at github. I am planning to do it after addressing all of your comments, so that I have simpler commit history.
Please let me know if you prefer to see the changes comment by comment.